src/core/strvector.c

```diff
@@ -182,25 +177,22 @@ igraph_error_t igraph_strvector_set(igraph_strvector_t *sv, igraph_integer_t idx
  */
 igraph_error_t igraph_strvector_set2(igraph_strvector_t *sv, igraph_integer_t idx,
                           const char *value, size_t len) {
-    if (idx < 0 || idx >= sv->len) {
+    char *tmp;
+
+    if (idx < 0 || idx >= sv->stor_end - sv->stor_begin) {
         IGRAPH_ERROR("String vector index out of bounds.", IGRAPH_EINVAL);
     }
```

Comment:
IMO this should be a fatal error (i.e. an `IGRAPH_ASSERT`) because we want this to fail hard in tests, and especially on OSS-fuzz. This is always a sign of a bug in the C code, so it should not be hidden when the error handler is set to "ignore".

Reply:
I've removed the check; we don't do this for vectors either, so if the goal is to make strvector more similar to the vector type, then we just leave it to the AddressSanitizer to do bounds checks.